bindings/rust/wasmedge-sys/src/error.rs

```diff
@@ -0,0 +1,47 @@
+use super::wasmedge::{
+    WasmEdge_Result, WasmEdge_ResultGetCode, WasmEdge_ResultGetMessage, WasmEdge_ResultOK,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct WasmEdgeError {
+    /// 0x00: Success
+    /// 0x01: Terminated -> Success
+    /// 0x02: Failed
+    /// 0x03: NullError
+    /// 0x20: File not found
+    pub code: u32,
+    pub message: String,
+}
+impl From<std::ffi::NulError> for WasmEdgeError {
+    fn from(e: std::ffi::NulError) -> WasmEdgeError {
+        WasmEdgeError {
+            code: 3,
+            message: e.to_string(),
+        }
+    }
+}
+impl From<WasmEdge_Result> for WasmEdgeError {
+    fn from(result: WasmEdge_Result) -> Self {
+        let code = unsafe { WasmEdge_ResultGetCode(result) };
+        let message = unsafe {
+            let c_str = std::ffi::CStr::from_ptr(WasmEdge_ResultGetMessage(result));
+            c_str.to_string_lossy().into_owned()
+        };
+        WasmEdgeError { code, message }
+    }
+}
+
+pub type WasmEdgeResult<T> = Result<T, WasmEdgeError>;
```

Comment:
WasmEdgeResult is a short form of Result<T, WasmEdgeError>, so that we can use WasmEdgeResult<_concrete_type_> instead of WasmEdgeResult<_concrete_type_, Error>. That would make the return type more concise. Thanks!